### ISSUE TITLE
Display CMIP6 Data Citation incl. license information

### ISSUE THREAD

**Issue by MartinaSt** [WCRP-CMIP/CMIP6_CVs#1062]
@durack1 and @matthew-mizielinski : As promised on the last WIP telco, I have created this issue to discuss my proposal further. 

Related to #1050 as a second step for displaying the license information, the data citation information should be added, as license and data citation are closely related. This consists of two steps:

1. The Data Citation service sets up an API providing the information on data citations.
2. The CMIP6 webpage created in #1050 should be extended by the rendered information from the API. A reference to the underlying API should be added.

The initial idea for the API content (response provided in JSON) is:

| DRS  | Project | Authors | Title | Publisher | Publication Year | DOI | ESGF_access_link | License | License Uri |  Model | Model Url |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |  ------------- |
| CMIP6.CMIP.MOHC.HadGEM3-GC31-LL | CMIP6 | Ridley, Jeff; Menary, Matthew; Kuhlbrodt, Till; Andrews, Martin; Andrews, Tim | MOHC HadGEM3-GC31-LL model output prepared for CMIP6 CMIP | Earth System Grid Federation | 2018 | https://doi.org/10.22033/ESGF/CMIP6.419 | http://esgf-data.dkrz.de/search/cmip6-dkrz/?mip_era=CMIP6&activity_id=CMIP&institution_id=MOHC&source_id=HadGEM3-GC31-LL | CC BY-SA 4.0 | http://creativecommons.org/licenses/by-sa/4.0/ | HadGEM3-GC31-LL | https://ukesm.ac.uk/licensing-of-met-office-nerc-and-niwa-cmip6-data/ | 	

The benefit would be an increased visibility and accessibility of data citation information. This goes back to a user suggestion from the CMIP6 survey. The content provided by the API overlaps with the license information in the source_id CV thus the license page could be rendered from the API information alone.

**Comment by MartinaSt** [comment]
@durack1 @matthew-mizielinski For the new HadGEM model UKESM1-1-LL the license information is missing: https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_source_id.json#L8250. 
Please add the missing information, so that every model description within the JSON includes the same information in the same format.

**Comment by MartinaSt** [comment]
@durack1 My first comment was too quick. Actually, license information is missing at least for AWI-ESM-2-1-LR (https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_source_id.json#L605), maybe more. It is much easier to pass a the source_id file, which has a consistent format and all models have the same information provided in the same JSON format, than having to deal with different JSON snippets.
Can you provide license_info for all models?

**Comment by matthew-mizielinski** [comment]
Hi @MartinaSt , the data for the models which have not published data (cohort of `Registered` rather than `Published`) was not present in the `CMIP6_source_id.json` file as there was no data to scrape this information from. We could put in a dictionary with blank values for each key, e.g.
```
            "license_info":{
                "exceptions_contact":"",
                "history":"",
                "id":"",
                "license":"",
                "source_specific_info":"",
                "url":""
            },
```
I did think about including a defaults for these values, but I think any information here could be problematic (e.g. if we chose to default to the `CC BY-SA 4.0` license and they really wanted the non-commercial version.

I'll throw in a pull request for this shortly.

**Comment by durack1** [comment]
@MartinaSt @matthew-mizielinski thanks for raising this. Matt has noted the issue, and has also solved this with blank entries in the PR https://github.com/WCRP-CMIP/CMIP6_CVs/pull/1070. I had planned to reach out to these "Registered" models (all 29 - 1, due to UKESM1-1-LL being published now) and ascertain their status before dealing with them - duplicated in the https://github.com/WCRP-CMIP/CMIP6_CVs/pull/1070#issuecomment-1137622383

**Comment by durack1** [comment]
@MartinaSt @matthew-mizielinski just to circle around, for both the `EC-Earth3*` and `IPSL*` models the license updates are now in place. For the models with no data published, the fields remain `"cohort": ["Registered"]` and the `"license_info": "history" = ""` as we have no version information to populate from the database. I have noted in #1076 and #1078 that reopening the issue (or opening a new issue) will be reqiured when these data are published

**Comment by durack1** [comment]
@MartinaSt did you still want to work on this issue, or should we close? If yes, then I suggest we create another HTML page that includes all your info, and link this to update (pulling info from your API) each time that an update is processed.

**Comment by MartinaSt** [comment]
@durack1 Yes, I still think we should do this, because it would make the CMIP6 data references visible. 
Updating that page after license updates in the CV should be sufficient. If you include me in the related github issues, I can ping you as soon as I have updated the information on my side.

**Comment by durack1** [comment]
@MartinaSt I believe we're very close to having finalized the license updates, in the sense that #1113 which captures the remaining tasks will be closed next week, with any remaining modeling groups that haven't relaxed left as they are.

For that reason you're in good shape to get this citation/data reference stuff ready to roll. Once we have #1113 closed, we can also propagate all the information back across the CMIP6 guide pages (ala https://github.com/orgs/WCRP-CMIP/projects/2) - will ping you on the #1113 issue so you know when this is closed

**Comment by durack1** [comment]
@MartinaSt it is not clear what steps are required to complete this issue? Did you want to augment information that is currently presented in [CMIP6_source_id_licenses.html](https://wcrp-cmip.github.io/CMIP6_CVs/docs/CMIP6_source_id_licenses.html), a new page alongside others, or a new page that integrates all the information presented across both [CMIP6_source_id_licenses.html](https://wcrp-cmip.github.io/CMIP6_CVs/docs/CMIP6_source_id_licenses.html) and [CMIP6_source_id.html](https://wcrp-cmip.github.io/CMIP6_CVs/docs/CMIP6_source_id.html)?

**Comment by MartinaSt** [comment]
@durack1 Let's restart the discussion and investigate the best option:

- The first decision is whether to include the citation information in the license page or not. The problem is that data citations are provided on two granularities. If I remember correctly, we favored to have a separate page with the data citations...?
- Secondly, how do we want to display the citation information? How about having a page with two tabs for the two granularities?
- Thirdly, how should we provide the information for you. There are some existing API available but we can easily set up an additional one. Existing options are:
    - Download a CSV from the CMIP6 citation search page http://bit.ly/CMIP6_Citation_Search (Actions>Download)
    - Use the API provided for ESGF DN managers https://www.wdc-climate.de/ui/cerarest/cmip6Citations?complete=true The disadvantage is that you might get some completed citation entries without data i.e. without DOIs.
    - Use DataCite's API: https://api.datacite.org/dois?query=publisher:Earth%20System%20Grid%20Federation The disadvantage is that you need to construct the citation information out of different metadata fields and exclude input4MIPs datasets.
    
There was a request/suggestion on what is needed from a user provided in the CMIP6 citation survey. If we need to create an additional API it should meet both needs.

**Comment by matthew-mizielinski** [comment]
@MartinaSt @durack1 

I've made some changes in #1177 to the code that generates the CMIP6_source_id_licenses.html that adds a DOI column with DOI links for each MIP/model/institution.  

I've picked up the second option @MartinaSt gave above (the cerarest api), which I had to break into a query for each model and strip information from (see pull request for details).

The resulting table looks a little cramped, but it does give us something to consider.

![image](https://user-images.githubusercontent.com/5744835/203068474-af3543c3-29ea-4e3f-93ac-a3a99bcce6be.png)

Does this look good enough or are there tweaks that would make this cleaner or clearer.

**Comment by matthew-mizielinski** [comment]
MartinaSt, could you confirm that I have the correct number of model level citations: 

I've got 558 citation DOIs via the ESGF DN api from the models that are registered in the CVs, and 536 dois are included in the table

The following 22 aren't used (i.e. no entry in CVs);

| entry | DOI link |
|------|------|
| CMIP6.CMIP.RTE-RRTMGP-Consortium.RTE-RRTMGP-181204 | https://doi.org/10.22033/ESGF/CMIP6.10125 |
| CMIP6.CMIP.MRI.MRI-AGCM3-2-H | https://doi.org/10.22033/ESGF/CMIP6.10943 |
| CMIP6.CMIP.NCAR.CESM1-WACCM-SC | https://doi.org/10.22033/ESGF/CMIP6.12282 |
| CMIP6.CMIP.IPSL.4AOP-v1-5 | https://doi.org/10.22033/ESGF/CMIP6.12341 |
| CMIP6.CMIP.IPSL.IPSL-CM6A-LR-INCA | https://doi.org/10.22033/ESGF/CMIP6.13582 |
| CMIP6.CMIP.IPSL.IPSL-CM5A2-INCA | https://doi.org/10.22033/ESGF/CMIP6.13642 |
| CMIP6.CMIP.AER.LBLRTM-12-8 | https://doi.org/10.22033/ESGF/CMIP6.2004 |
| CMIP6.CMIP.EC-Earth-Consortium.EC-Earth3P-VHR | https://doi.org/10.22033/ESGF/CMIP6.2326 |
| CMIP6.CMIP.AER.RRTMG-LW-4-91 | https://doi.org/10.22033/ESGF/CMIP6.9962 |
| CMIP6.CMIP.AER.RRTMG-SW-4-02 | https://doi.org/10.22033/ESGF/CMIP6.9964 |

The following I'm just ignoring as test entries
| entry | DOI link |
|------|------|
| CMIP6.CORDEX.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.1166 |
| CMIP6.DynVarMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.1167 |
| CMIP6.SIMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.1168 |
| CMIP6.VIACSAB.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.1169 |
| CMIP6.C4MIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.843 |
| CMIP6.HighResMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.848 |
| CMIP6.DCPP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.850 |
| CMIP6.LUMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.851 |
| CMIP6.AerChemMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.853 |
| CMIP6.PMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.856 |
| CMIP6.OMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.857 |
| CMIP6.ScenarioMIP.PCMDI.PCMDI-test-1-0 | https://doi.org/10.5072/ESGF/CMIP6.858 |

**Comment by matthew-mizielinski** [comment]
Moving a conversation from a commit to the issue;

> [MartinaSt](https://github.com/MartinaSt) commented on [687a199](https://github.com/WCRP-CMIP/CMIP6_CVs/commit/687a199803a000ff384c888ee583767f5cc1981a) [yesterday](https://github.com/WCRP-CMIP/CMIP6_CVs/commit/687a199803a000ff384c888ee583767f5cc1981a#commitcomment-90868165)
> Thanks @matthew-mizielinski !
> I am afraid we are not done.
> 
> I. The API Request/Response:
> The API for the ESGF DN Managers has two problems:
> 
> Option complete="true" means the completeness of the citation information or the readiness for ESGF publication. Therefore, you will get a lot of entries, without ESGF data and without DOIs in your response.
> The response provides the "reserved DOI" without checking this is actually registered. After the WGI AR6 literature and data submission deadline, a couple of datasets have been published in the ESGF without completing the citation information. Thus, the draft DOIs for these few were never registered and the draft DOIs in your table will not resolve.
> The response is limited to 1999. You have to send separate requests for each institution_id (option "institutionId") to get the full list.
> II. The Table column
> I would name the column "data reference" and display the commonly used short form of citations "first author et al. (year)" with the DOI link.
> 
> How to continue...
> As I got a request to provide a flexible API to extract data citation information from the database and as I do not want to change the existing API, we will set up a dedicated Citation Search API providing the data references but also the individual components of the data references and with several request options. I have created a database view for the REST API with the columns:
> DRS
> MIP_ERA
> ACTIVITY_ID
> INSTITUTION_ID
> SOURCE_ID
> EXPERIMENT_ID
> GRANULARITY ([exp|model] allows for granularity filtering)
> AUTHORS
> SHORT_AUTHORS (see above)
> TITLE
> PUBLISHER
> PUBLICATION_YEAR
> DOI
> DATA_REFERENCE (as provided by the Citation Search GUI)
> ESGF_ACCESS_LINK
> LICENSE
> LICENSE_URI
> MODEL_URL
> 
> All columns can be possible API options. My suggestion for options.
> MIP_ERA
> ACTIVITY_ID
> INSTITUTION_ID
> SOURCE_ID
> EXPERIMENT_ID
> GRANULARITY
> Do you think these filter options are sufficient?
> Note: The DRS component names are valid for CMIP6, for input4MIPs these are wrong. But I have to name the columns somehow.[MartinaSt](https://github.com/MartinaSt) commented on [687a199](https://github.com/WCRP-CMIP/CMIP6_CVs/commit/687a199803a000ff384c888ee583767f5cc1981a) [yesterday](https://github.com/WCRP-CMIP/CMIP6_CVs/commit/687a199803a000ff384c888ee583767f5cc1981a#commitcomment-90868165)
> Thanks @matthew-mizielinski !
> I am afraid we are not done.
> 
> I. The API Request/Response:
> The API for the ESGF DN Managers has two problems:
> 
> Option complete="true" means the completeness of the citation information or the readiness for ESGF publication. Therefore, you will get a lot of entries, without ESGF data and without DOIs in your response.
> The response provides the "reserved DOI" without checking this is actually registered. After the WGI AR6 literature and data submission deadline, a couple of datasets have been published in the ESGF without completing the citation information. Thus, the draft DOIs for these few were never registered and the draft DOIs in your table will not resolve.
> The response is limited to 1999. You have to send separate requests for each institution_id (option "institutionId") to get the full list.
> II. The Table column
> I would name the column "data reference" and display the commonly used short form of citations "first author et al. (year)" with the DOI link.
> 
> How to continue...
> As I got a request to provide a flexible API to extract data citation information from the database and as I do not want to change the existing API, we will set up a dedicated Citation Search API providing the data references but also the individual components of the data references and with several request options. I have created a database view for the REST API with the columns:
> DRS
> MIP_ERA
> ACTIVITY_ID
> INSTITUTION_ID
> SOURCE_ID
> EXPERIMENT_ID
> GRANULARITY ([exp|model] allows for granularity filtering)
> AUTHORS
> SHORT_AUTHORS (see above)
> TITLE
> PUBLISHER
> PUBLICATION_YEAR
> DOI
> DATA_REFERENCE (as provided by the Citation Search GUI)
> ESGF_ACCESS_LINK
> LICENSE
> LICENSE_URI
> MODEL_URL
> 
> All columns can be possible API options. My suggestion for options.
> MIP_ERA
> ACTIVITY_ID
> INSTITUTION_ID
> SOURCE_ID
> EXPERIMENT_ID
> GRANULARITY
> Do you think these filter options are sufficient?
> Note: The DRS component names are valid for CMIP6, for input4MIPs these are wrong. But I have to name the columns somehow.

thanks @MartinaSt,  

The code I put in was running a separate request for each source_id in order to avoid the limit on the number of records returned as I realised that was constraining things, but I didn't know anything else about the limitations of this service. I had seen a few dois that didn't resolve , but it didn't occur to me what might be going on. Thanks for the explanation

@durack1 and I had chat about the first draft I threw together and I think the approach I took to add a column to an existing table is getting too busy so I was going to refactor this to produce a new table, at which point we can add more information.

I think a separate table can contain more than just a single cell per model so I think the following would be a good next iteration 

One table for the per model DOIs

| Source ID | Institution ID | Activity ID | Data Reference | DOI | 
|----|----|---|---|---|
| UKESM1-0-LL |  MOHC | AerChemMIP | `O'Connor, Fiona (2019). MOHC UKESM1.0-LL model output prepared for CMIP6 AerChemMIP. Earth System Grid Federation. doi:https://doi.org/10.22033/ESGF/CMIP6.1561` | [doi:10.22033/ESGF/CMIP6.1561](https://doi.org/10.22033/ESGF/CMIP6.1561) |
| ... | ... | ... | ... | ... | 

and for a per experiment reference;

| Source ID | Institution ID | Activity ID | Experiment ID | Data Reference | DOI| 
|----|----|---|---|---|---|
| UKESM1-0-LL |  MOHC | AerChemMIP | histSST| `O'Connor, Fiona (2018). MOHC UKESM1.0-LL model output prepared for CMIP6 AerChemMIP histSST. Earth System Grid Federation. doi:https://doi.org/10.22033/ESGF/CMIP6.6077` | [doi:10.22033/ESGF/CMIP6.6077](https://doi.org/10.22033/ESGF/CMIP6.6077) |
| ... | ... | ... | ... | ... | ... | 

I think the bare minimum queries I'd need to do for this based on your suggestion for the new API are;

`base_url?mip_era=CMIP6&sourceID=<source_id>&granularity=model` for each source_id in the CVs

which I think would typically give up to 20 records per query, with ~100 queries total.

For the per experiment table I think I'd need to break it down further (UKESM1-0-LL has 70-80 experiments)
`base_url?mip_era=CMIP6&sourceID=<source_id>&activityId=<activity_id>&granularity=experiment` for each (source_id, activity_id) combination in the CVs

**Comment by MartinaSt** [comment]
@matthew-mizielinski Thanks for the explanation. Then your list of references is complete.

I don't really like the approach to separate the data citation from the license because 
1. the displayed licenses are licenses of the data not of the model. For software other licenses are used; 
2. the CC BY licenses say that the creator should be acknowledged, thus the license is directly  related to the data citation.

So, I still favor the integration of short form data references (first author et al. (year)) into the existing license table. In case you want to follow the separate table approach, we need to make my two points above clear on the existing page and refer to the page containing data citation.  

My suggestions for the new data citation pages:
- add the license 
- add model url
- add export of citation in common formats like bibtex, RIS
- use the abbreviated authors because some institutions provided >40 authors

Your example API requests will be supported by the planned new API. I have additionally added a column/API option to support updates of the tables: Additional column 'PUBLICATION_DATE' and an option to get DOIs published after a given date. 

My colleague will work on the REST API next week. The option for a citation export in bibtex,RIS format won't be supported in the first version but is planned for a later version. For the data citation tables, we can include DataCite's API links for the exports in bibtex,RIS formats. These will be maintained on the long-term, whereas DKRZ's level of curation for these APIs will be limited. For your example, you can find the links on DataCite's commons page: https://commons.datacite.org/doi.org/10.22033/esgf/cmip6.1561:
- bibtex: https://api.datacite.org/application/x-bibtex/10.22033/esgf/cmip6.1561
- RIS: https://api.datacite.org/application/x-research-info-systems/10.22033/esgf/cmip6.1561

**Comment by MartinaSt** [comment]
@matthew-mizielinski The endpoint for the new API is http://cera-www.dkrz.de/WDCC/ui/cerasearch/cerarest/cmip6search

Available filter options:
- filter by DRS:
  - mipEra
  - activityId
  - institutionId
  - sourceId
  - experimentId
- filter by granularity:
  - granularity=[exp|model]
- filter by date (in ISO 8601 format):
  - gePublicationDate=YYYY-MM-DD: DOI published at or after a given date
  - lePublicationDate=YYYY-MM-DD: DOI published before or at a given date
  
Sample Calls:
> base_url?mip_era=CMIP6&sourceID=<source_id>&granularity=model for each source_id in the CVs  
http://cera-www.dkrz.de/WDCC/ui/cerasearch/cerarest/cmip6search?mipEra=CMIP6&sourceId=HadGEM3-GC31-MM&granularity=model

> base_url?mip_era=CMIP6&sourceID=<source_id>&activityId=<activity_id>&granularity=experiment for each (source_id, activity_id) combination in the CVs
http://cera-www.dkrz.de/WDCC/ui/cerasearch/cerarest/cmip6search?mipEra=CMIP6&activityId=CMIP&sourceId=HadGEM3-GC31-MM&granularity=exp

Update of this request:
http://cera-www.dkrz.de/WDCC/ui/cerasearch/cerarest/cmip6search?mipEra=CMIP6&activityId=CMIP&sourceId=HadGEM3-GC31-MM&granularity=exp&gePublicationDate=2020-01-01

Data references on model/MIP (coarse) granularity contributing to an activity_id available at a given snap-shot date:
http://cera-www.dkrz.de/WDCC/ui/cerasearch/cerarest/cmip6search?mipEra=CMIP6&activityId=ScenarioMIP&granularity=model&lePublicationDate=2020-03-31

**Comment by MartinaSt** [comment]
@matthew-mizielinski, thanks, I could check the citation-source_id table, the one for the experiment_id was too long. It looks good.

One request: Could you add the license in an additional column?

In case the full citation string gets to long for some entries and messes up the table, you could construct the citation string out of the components in the JSON using the SHORT_AUTHORS.

**Comment by matthew-mizielinski** [comment]
> @matthew-mizielinski, thanks, I could check the citation-source_id table, the one for the experiment_id was too long. It looks good.

Yes, 2000 records is a little much for viewing from github -- you'll either need to check out the repo to see this or you can take my assurance that it is very similar to the other table that you can easily view.

> 
> One request: Could you add the license in an additional column?
> 

Done.

> In case the full citation string gets to long for some entries and messes up the table, you could construct the citation string out of the components in the JSON using the SHORT_AUTHORS.

I think this is clearer, so I've done this too.

**Comment by durack1** [comment]
Closed by #1177

**Comment by durack1** [comment]
@matthew-mizielinski @MartinaSt let me know if we need to augment the README.md or the CMIP6 website to add links to these new resources - if yes, maybe reopen this issue
